src/fetch_request.js

```diff
@@ -97,12 +98,15 @@ export class FetchRequest {
     const originalQuery = (this.originalUrl.split('?')[1] || '').split('#')[0]
     const params = new URLSearchParams(originalQuery)
 
-    if (this.options.query) {
-      for (const [key, value] of Object.entries(this.options.query)) {
-        params.append(key, value)
-      }
+    let requestQuery = this.options.query
+    if (requestQuery instanceof window.FormData) {
+      requestQuery = stringEntriesFromFormData(requestQuery)
+    } else {
+      requestQuery = Object.entries(requestQuery || {})
```

Comment:
Do you know if this can handle `URLSearchParams` objects too? It might already and it would be nice to support objects, URLSearchParams, and FormData.

Reply:
Just tested and it doesn't seem so, but URLSearchParams does have an entries method. https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/entries

I guess we could do:

```javascript
if (requestQuery instanceof window.FormData) {
  requestQuery = stringEntriesFromFormData(requestQuery)
} else if (requestQuery instanceof window.URLSearchParams) {
  requestQuery = requestQuery.entries()
} else {
  requestQuery = Object.entries(requestQuery || {})
}
```